pkg/cluster/cluster.go

```diff
@@ -99,36 +100,60 @@ func new(config Config, s *spec.ClusterSpec, stopC <-chan struct{}, wg *sync.Wai
 		status:  &Status{},
 	}
 
-	backup := c.spec.Backup
-	if backup != nil {
-		err := k8sutil.CreateBackupReplicaSetAndService(c.KubeCli, c.Name, c.Namespace, c.PVProvisioner, *backup)
-		if err != nil {
-			return nil, fmt.Errorf("fail to create backup: %v", err)
-		}
+	if err := c.prepareBackupAndRestore(); err != nil {
+		return nil, err
+	}
+	if err := c.prepareSeedMember(isNewCluster); err != nil {
+		return nil, err
 	}
 
 	if err := c.createClientServiceLB(); err != nil {
 		return nil, fmt.Errorf("fail to create client service LB: %v", err)
 	}
-	if isNewCluster {
-		var err error
-		if c.spec.SelfHosted != nil {
-			if len(c.spec.SelfHosted.BootMemberClientEndpoint) == 0 {
-				err = c.newSelfHostedSeedMember()
-			} else {
-				err = c.migrateBootMember()
-			}
-		} else {
-			err = c.newSeedMember()
+
+	wg.Add(1)
+	go c.run(stopC, wg)
+
+	return c, nil
+}
+
+func (c *Cluster) prepareBackupAndRestore() error {
+	backup := c.spec.Backup
+	restore := c.spec.Restore
+	switch {
+	case backup == nil && restore != nil:
+		return errors.New("restore policy set but backup policy not. rejected. dead")
+	case backup != nil && restore == nil:
+		err := k8sutil.CreateAndWaitPVC(c.KubeCli, c.Name, c.Namespace, c.PVProvisioner, backup.VolumeSizeInMB)
+		if err != nil {
+			return err
 		}
+		fallthrough
+	case backup != nil && restore != nil:
+		err := k8sutil.CreateBackupReplicaSetAndService(c.KubeCli, c.Name, c.Namespace)
```

Comment:
add a todo here? we now claim the original pv whatever the current cluster name is. we should do a copy if the name is different.